src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,77 @@
+package duke;
+
+import java.io.IOException;
+
+public class Duke {
+    private final Ui ui;
+    private final Parser parser;
+    String dest;
+
+    /**
+     * Constructor with specified directory of database.
+     *
+     * @param dest directory of database
+     */
+    public Duke(String dest) {
+        this.dest = dest;
+        ui = new Ui();
+        parser = new Parser();
+    }
+
+    private void run() {
+        Storage storage;
+        TaskList taskList;
+        try {
+            storage = new Storage(dest);
+            taskList = storage.loadFile();
+        } catch (IOException e) {
+            ui.printNicely(e.getMessage());
+            return;
+        }
+        ui.greet();
+        ui.listOut(taskList);
+        while (true) {
```

Comment:
Try to avoid while (true) ? I think its generally bad practice? Can replace with boolean flags

Reply:
Resolved.